hack/generator/pkg/codegen/code_generator.go

```diff
@@ -145,23 +157,37 @@ func (loader *cancellableJSONLoader) LoaderFactory() gojsonschema.JSONLoaderFact
 	return &cancellableJSONLoaderFactory{loader.ctx, loader.inner.LoaderFactory()}
 }
 
-func loadSchema(ctx context.Context, idFactory astmodel.IdentifierFactory, configuration *config.Configuration) PipelineStage {
+type schemaLoader func(ctx context.Context, source string) (*gojsonschema.Schema, error)
+
+func defaultSchemaLoader(ctx context.Context, source string) (*gojsonschema.Schema, error) {
+	sl := gojsonschema.NewSchemaLoader()
+	loader := &cancellableJSONLoader{
+		ctx,
+		gojsonschema.NewReferenceLoaderFileSystem(source, &cancellableFileSystem{ctx}),
+	}
+
+	schema, err := sl.Compile(loader)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error loading schema from %q", source)
+	}
+
+	return schema, nil
+}
+
+func loadSchemaIntoTypes(
```

Comment:
while you’re here do you want to move this PipelineStage generator function and all its dependencies into another file? it has a lot of cruft around it for cancellation

Reply:
sure

Reply:
@matthchr feels nicer 👍